File: backend/src/constants/teams.go
package constants

// TeamInfo binds a team id to its display label and the task types it serves.
// Task generation chooses the owning team per GroundTaskType through this table.
type TeamInfo struct {
	ID        string
	Name      string
	TaskKinds []GroundTaskType
}

var Teams = []TeamInfo{
	{"TEAM-CLEAN", "客舱清洁班", []GroundTaskType{TaskCleaning}},
	{"TEAM-CATER", "航机配餐班", []GroundTaskType{TaskCatering}},
	{"TEAM-BAG", "行李装卸班", []GroundTaskType{TaskBaggage}},
	{"TEAM-FUEL", "航油保障班", []GroundTaskType{TaskRefuel}},
	{"TEAM-WATER", "清水污水班", []GroundTaskType{TaskWaterService}},
	{"TEAM-RAMP", "机坪牵引班", []GroundTaskType{TaskPushback}},
}

// TeamForTask returns the owning team for a task type.
func TeamForTask(kind GroundTaskType) TeamInfo {
	for _, team := range Teams {
		for _, served := range team.TaskKinds {
			if served == kind {
				return team
			}
		}
	}
	return Teams[0]
}

func TeamText(teamID string) string {
	for _, team := range Teams {
		if team.ID == teamID {
			return team.Name
		}
	}
	return teamID
}

// ResourceTypeForTask maps a task type to the resource pool it consumes.
func ResourceTypeForTask(kind GroundTaskType) string {
	return string(kind)
}

File: frontend/src/constants/teams.ts
import { GROUND_TASK_TYPE_TEXT, type GroundTaskTypeValue } from "./GroundTaskType";

export interface TeamInfo {
  id: string;
  name: string;
  taskKinds: GroundTaskTypeValue[];
}

export const TEAMS: TeamInfo[] = [
  { id: "TEAM-CLEAN", name: "客舱清洁班", taskKinds: ["CLEANING"] },
  { id: "TEAM-CATER", name: "航机配餐班", taskKinds: ["CATERING"] },
  { id: "TEAM-BAG", name: "行李装卸班", taskKinds: ["BAGGAGE"] },
  { id: "TEAM-FUEL", name: "航油保障班", taskKinds: ["REFUEL"] },
  { id: "TEAM-WATER", name: "清水污水班", taskKinds: ["WATER_SERVICE"] },
  { id: "TEAM-RAMP", name: "机坪牵引班", taskKinds: ["PUSHBACK"] }
];

export function teamText(teamId: string): string {
  return TEAMS.find((t) => t.id === teamId)?.name ?? teamId;
}

export function taskTypeText(value: string): string {
  return GROUND_TASK_TYPE_TEXT[value as GroundTaskTypeValue] ?? value;
}
